sections/header.liquid

```diff
@@ -458,18 +464,33 @@
       observer.observe(this.header);
     }
 
+    setPreventReveal() {
+      this.preventReveal = true;
+    }
+
     onScroll() {
       const scrollTop = window.pageYOffset || document.documentElement.scrollTop;
 
-      if (scrollTop > this.currentScrollTop && scrollTop > this.headerBounds.bottom) {
+      window.clearTimeout(this.isScrolling);
+
+      this.isScrolling = setTimeout(() => {
+        this.preventReveal = false;
+      }, 66);
+
+      if (this.preventReveal) {
         requestAnimationFrame(this.hide.bind(this));
-      } else if (scrollTop < this.currentScrollTop && scrollTop > this.headerBounds.bottom) {
-        requestAnimationFrame(this.reveal.bind(this));
-      } else if (scrollTop <= this.headerBounds.top) {
-        requestAnimationFrame(this.reset.bind(this));
-      }
+        this.currentScrollTop = scrollTop;
```

Comment:
This line is both on the `if` and on the `else` blocks. Does it need to happen AFTER the `requestAnimationFrame`? If so, ignore this suggestion.

If not, move it outside the `if`/`else` block.

Also, **nitpick:** an early return could probably help with legibility.